### PR TITLE
WIP on getting app-prop-viewer to load. Seems to get to component now…

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
* Fixed build problems. Environment object was missing
* Add `ng build` to the package.json build instruction so you can `npm run build` because I never put npm tools in my global path and we dont tell people to either.
* Remove "component factory" code of the angular-2-plugin-loader that was never used as far as I know (the "component.js" content)
* Removed code about html2canvas. Dead code should not be kept around.
* Update app-prop-viewer to newer versions of libraries
* Update app-prop-viewer to webpack5 base config
* Move externals requirejs loading from the externals.js file into the desktop.js file
Before this, the app would try to do a GET on `web/@angular/core/core.js`
After, it makes no request at all.
Neither does desktop.js. The "externals" seem to work in the same way they always have, but now without the need for a separate "externals.js" and "externals-main.js"

NOTE: `ng build` removes `require.js` from the web folder. After building you still need to `cp node_modules/requirejs/require.js web/`...

### CODE REVIEW

File: virtual-desktop/angular.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
               "sourceMap": true
             }
           },
-          "defaultConfiguration": "production"
+          "defaultConfiguration": "development"
```
Comment (DivergentEuropeans) [inline virtual-desktop/angular.json:59]: What's this for?

File: virtual-desktop/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
   "scripts": {
     "build:externals": "webpack --progress --config webpack.externals.js",
     "start": "webpack --watch --progress",
```
Comment (DivergentEuropeans) [inline virtual-desktop/package.json:11]: You can also update start script to be
"ng build --watch"

**Comment by DivergentEuropeans** [comment]
I am getting this, is this expected UI? 

image

The console is very promising, some App Prop Viewer things are being loaded but not sure if I'm seeing what I'm supposed to

**Comment by DivergentEuropeans** [comment]
small addition: https://github.com/zowe/zlux-app-manager/pull/606

File: virtual-desktop/package.json (hunk 2 above)
Comment (DivergentEuropeans) [reply virtual-desktop/package.json:11]: added

File: virtual-desktop/angular.json (hunk 1 above)
Comment (1000TurquoisePogs) [reply virtual-desktop/angular.json:59]: development, i need to see the source in the browser to debug.
it can be production when we're ready.

**Comment by 1000TurquoisePogs** [comment]
> I am getting this, is this expected UI?
> 
> image
> 
> The console is very promising, some App Prop Viewer things are being loaded but not sure if I'm seeing what I'm supposed to

yes, but i dont know why it doesnt get further. perhaps some step is missing.

**Pull request closed without merging**